plugins/contactpressuresensorsarray/src/ContactPressureSensorsArrayDriver.cpp

```diff
@@ -0,0 +1,516 @@
+/*
+ * Copyright (C) 2018 Fondazione Istituto Italiano di Tecnologia RBCS & iCub Facility & ADVR
+ * Authors: see AUTHORS file.
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or any later version, see LGPL.TXT or LGPL3.TXT
+ */
+
+#include "ContactPressureSensorsArrayDriver.h"
+#include <GazeboYarpPlugins/Handler.hh>
+#include <GazeboYarpPlugins/common.h>
+
+#include <yarp/os/LockGuard.h>
+#include <yarp/os/Log.h>
+#include <yarp/os/LogStream.h>
+#include <yarp/math/Math.h>
+#include <yarp/math/SVD.h>
+
+#include <sdf/Param.hh>
+
+#define DEBUG 0
+#define N_AXIS 3
+#define TIME_CONVERSION_NS 1.0e-9
+
+using namespace yarp::dev;
+
+GazeboYarpContactPressureSensorsArrayDriver::GazeboYarpContactPressureSensorsArrayDriver()
+{
+
+}
+
+GazeboYarpContactPressureSensorsArrayDriver::~GazeboYarpContactPressureSensorsArrayDriver()
+{
+
+}
+
+void GazeboYarpContactPressureSensorsArrayDriver::onUpdate(const gazebo::common::UpdateInfo& _info)
+{ 
+    gazebo::msgs::Contacts contacts;
+    contacts = this->m_sensor->Contacts();
```

Comment:
I guess this involves a copy of quite a big object. Have you considered using a const reference for this?

Reply:
Actually I hadn't put enough thought into the memory perspective... Thanks for opening my eyes. I think this could help improve the was-failing RTF.

Reply:
Apparently, with the contact sensor plugin, the real time factor drops down to around 0.12 and without the plugin, real time factor stays around 0.69